File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        // make a stack to make sure everytime you open one you are closing
        
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            
            if (s.charAt(i) == '(') {
                stack.push(')');
                continue;
            } else if (s.charAt(i) == '[') {
                stack.push(']');
                continue;
            } else if (s.charAt(i) == '{') {
                stack.push('}');
                continue;
            }

            if (stack.isEmpty() || s.charAt(i) != stack.peek()) {
                return false;
            }

            stack.pop();
            

        }

        return stack.isEmpty();

    }
}

File: Data Structures & Algorithms/last-stone-weight/submission-1.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        if not stones:
            return 0

        stones.sort()
        while len(stones) > 1:
            print(stones)
            if stones[-1] == stones[-2]:
                stones.pop()
                stones.pop()
            elif stones[-1] > stones[-2]:
                stones[-2] = stones[-1] - stones[-2]
                stones.pop()
            stones.sort()
        if not stones:
            return 0
        return stones[0]